pkg/plugin/plugin.go

```diff
@@ -38,9 +38,32 @@ type Base interface {
 	SupportedProjectVersions() []string
 }
 
-// Key returns a Base plugin's unique identifying string.
-func Key(p Base) string {
-	return path.Join(p.Name(), "v"+strings.TrimLeft(p.Version(), "v"))
+// KeyFor returns a Base plugin's unique identifying string.
+func KeyFor(p Base) string {
+	return Key(p.Name(), p.Version())
+}
+
+// Key returns a unique identifying string for a plugin's name and version.
+func Key(name, version string) string {
+	if version == "" {
+		return name
+	}
+	return path.Join(name, "v"+strings.TrimLeft(version, "v"))
+}
+
+// KeyFrom returns a name and version for an arbitrary key.
+func KeyFrom(key string) (string, string) {
+	if !strings.Contains(key, "/") {
+		return key, ""
+	}
+	keyParts := strings.Split(key, "/")
+	return keyParts[0], keyParts[1]
+}
+
+// GetShortName returns plugin's short name (name before domain) if name
+// is fully qualified (has a domain suffix), otherwise GetShortName returns name.
+func GetShortName(name string) string {
+	return strings.SplitN(name, ".", 2)[0]
```

Comment:
No need to make `SplitN` keep splitting after the first dot if we only want the first part, even if every domain needs to have at least 2 parts.

```suggestion
	return strings.SplitN(name, ".", 1)[0]
```

Reply:
[`SplitN`](https://golang.org/pkg/strings/#SplitN) will return a slice with at most `n` substrings, so `n = 1` will give you the whole name no matter how many `"."` exist in `name`.

Reply:
You are right, you can mark this as resolved, will update the other.